src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1829,7 +1811,7 @@ fn add_upstream_rust_crates<'a, B: ArchiveBuilder<'a>>(
 
             let mut any_objects = false;
             for f in archive.src_files() {
-                if f.ends_with(RLIB_BYTECODE_EXTENSION) || f == METADATA_FILENAME {
+                if f == METADATA_FILENAME {
```

Comment:
Yay, one less reason to rewrite rlibs every time we link. I wonder if there is a way for the linker to skip the metadata file without having to remove it from the archive.

Reply:
That is not true, rlibs are already passed directly to the linker. Only in situations like LTO where we want to remove object code or we're doing something `-force_load` do we remove things from rlibs.

Reply:
Missed the return above.